### PR TITLE
Fix 'queueMessages' expected behaviour

### PR DESCRIPTION
**Pull request by ricardopereira**
#1004

### CODE REVIEW

File: Spec/RealtimeClientChannel.swift [hunk 1]
```diff
@@ -496,6 +496,82 @@ class RealtimeClientChannel: QuickSpec {
...
+                        options.queueMessages = false
+                        options.autoConnect = false
+                        let client = ARTRealtime(options: options)
+                        defer { client.dispose(); client.close() }
+                        let channel = client.channels.get("foo")
+                        waitUntil(timeout: testTimeout) { done in
+                            expect(client.connection.state).to(equal(.initialized))
+                            channel.publish(nil, data: "message") { error in
+                                expect(error).toNot(beNil())
+                                done()
```
Comment (ricardopereira) [inline Spec/RealtimeClientChannel.swift]: @paddybyers The publish should fail if the connection is in the INITIALIZED state, right? [`TO3g`](https://docs.ably.io/client-lib-development-guide/features/#TO3g) only mentions about connecting but it contradicts with [`RTL6c2`](https://docs.ably.io/client-lib-development-guide/features/#RTL6c2).

Comment (ricardopereira) [reply Spec/RealtimeClientChannel.swift]: cc @QuintinWillison

**Comment by QuintinWillison** [comment]
**Breadcrumbs**: [internal discussion](https://ably-real-time.slack.com/archives/C030C5YLY/p1585300078019400) around failing tests which may be blocking @ricardopereira from fully testing this, which may be why this PR is still in Draft state.

**Comment by ricardopereira** [comment]
Yes, exactly. Thanks @QuintinWillison.

**Comment by ricardopereira** [comment]
The CI is getting better but even so I had to restart some of them.

**Comment by ricardopereira** [comment]
@QuintinWillison BTW, this fixes https://github.com/ably/ably-cocoa/issues/894 too 🙌

File: Source/ARTRealtime.m [hunk 2]
```diff
@@ -699,42 +701,53 @@ - (ARTEventListener *)transitionSideEffects:(ARTConnectionStateChange *)stateCha
...
+                    [channel detachChannel:[ARTStatus state:ARTStateOk]];
+                    break;
+                case ARTRealtimeSuspended:
+                    [channel setSuspended:channelStatus];
+                    break;
+                case ARTRealtimeFailed:
+                    [channel setFailed:channelStatus];
+                    break;
+                default:
+                    break;
```
Comment (SimonWoolf) [inline Source/ARTRealtime.m:750]: Why is all of this still gated under a condition that fails if `options.queueMessages` is false? This is [RTL3](https://sdk.ably.com/builds/ably/specification/main/features/#RTL3), right? That spec item doesn't mention anything about queueMessages, what do they have to do with each other?

Comment (maratal) [reply Source/ARTRealtime.m:750]: See https://github.com/ably/ably-cocoa/pull/1941

**Pull request merged**